pkg/retry/retry.go

```diff
@@ -5,9 +5,199 @@
 
 package retry
 
-import "time"
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/pkg/errors"
+)
 
 const (
 	DefaultLinearBackoffInterval = time.Second
 	DefaultLinearRetryCount      = 3
 )
+
+//RetryStrategy is the type that represents Retry Policy settings
+type RetryStrategy string
+
+const (
+	off                           RetryStrategy = "off"
+	linear                        RetryStrategy = "linear"
+	exponential                   RetryStrategy = "exponential"
+	defaultRetryStrategy          RetryStrategy = linear
+	defaultRetryMaxCount          int           = 3
+	defaultRetryIntervalInSeconds int           = 1
+	MinRetryMaxCount              int           = 1
+	MaxRetryMaxCount              int           = 5
+	MinRetryIntervalInSeconds     int           = 1
+	MaxRetryIntervalInSeconds     int           = 3
+)
+
+var log = logger.NewLogger("dapr.runtime.retry")
```

Comment:
Get the logger from the caller, please. It will give better context to what the retry is about.

Reply:
Fixed